pkg/controllers/membercluster/v1beta1/membercluster_controller.go

```diff
@@ -154,7 +154,7 @@ func (r *Reconciler) handleDelete(ctx context.Context, mc *clusterv1beta1.Member
 			return runtime.Result{}, controller.NewAPIServerError(true, err)
 		}
 		// we don't need to wait for the agent to leave if the internal member cluster is not found
-		if condition.IsConditionStatusTrue(cond, mc.GetGeneration()) {
+		if condition.IsConditionStatusTrue(mcJoinedCondition, mc.GetGeneration()) {
```

Comment:
now it becomes expected, we garbage collect in line 173 and the update mc call could fail in line 176.

i'm thinking of skipping updating the mc condition to simplify the implementation here?